caf-logging-tomcat-juli/pom.xml

```diff
@@ -0,0 +1,227 @@
+                                </artifactItem>
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>3.2.1</version>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <createDependencyReducedPom>false</createDependencyReducedPom>
+                            <createSourcesJar>true</createSourcesJar>
+                            <shadeSourcesContent>true</shadeSourcesContent>
+                            <artifactSet>
+                                <includes>
+                                    <include>org.apache.tomcat:tomcat-juli</include>
+                                    <include>org.slf4j:jcl-over-slf4j</include>
+                                    <include>org.slf4j:slf4j-api</include>
+                                    <include>ch.qos.logback:logback-classic</include>
+                                    <include>ch.qos.logback:logback-core</include>
+                                    <include>com.fasterxml.jackson.core:jackson-core</include>
+                                    <include>org.apache.commons:commons-text</include>
+                                    <include>com.github.cafapi.logging:caf-logging-logback</include>
+                                    <include>com.github.cafapi.util:util-process-identifier</include>
+                                    <include>com.github.cafapi.logging:caf-logging-common</include>
+                                </includes>
+                            </artifactSet>
+
+                            <filters>
+                                <filter>
+                                    <artifact>org.apache.tomcat:tomcat-juli</artifact>
+                                    <includes>
+                                        <include>org/apache/juli/ClassLoaderLogManager**</include>
+                                        <include>org/apache/juli/WebappProperties.class</include>
+                                        <include>META-INF/LICENSE</include>
+                                        <include>META-INF/NOTICE</include>
+                                    </includes>
+                                </filter>
+                                <filter>
+                                    <artifact>com.github.cafapi.logging:caf-logging-logback</artifact>
+                                    <excludes>
+                                        <exclude>logback.xml</exclude>
+                                    </excludes>
```

Comment:
Adding excludes is fine when there is no other choice, but if this is required then it suggests to me that what we should actually do is to split `caf-logging-logback` up, i.e. into `caf-logging-logback` and `caf-logging-logback-converters`, so that it is possible to use the converters with a completely custom logback.xml file.  In fact perhaps we should go further and have separate converter modules for each one (I wonder whether [caf-logging-logback-access/.../ProcessAndThreadIdPatternConverter.java](https://github.com/CAFapi/caf-logging/blob/master/caf-logging-logback-access/src/main/java/com/github/cafapi/logging/logback/access/ProcessAndThreadIdPatternConverter.java) would actually be required if [caf-logging-logback/.../ProcessAndThreadIdPatternConverter.java](https://github.com/CAFapi/caf-logging/blob/master/caf-logging-logback/src/main/java/com/github/cafapi/logging/logback/ProcessAndThreadIdPatternConverter.java) was available standalone).

Reply:
> converters

I try to use the  [caf-logging-logback-converters/.../ProcessAndThreadIdPatternConverter.java](https://github.com/CAFapi/caf-logging/blob/master/caf-logging-logback-converters/src/main/java/com/github/cafapi/logging/logback/converters/ProcessAndThreadIdPatternConverter.java) Inside logback-access.xml for threadId. But I was getting below exception.

0:48:41.251 documentid-service> 10:48:41,207 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.access.PatternLayoutEncoder] for [encoder] property
10:48:41.254 documentid-service> 10:48:41,211 |-ERROR in ch.qos.logback.core.pattern.parser.Compiler@26204f8f - Failed to instantiate converter class [com.github.cafapi.logging.logback.converters.ProcessAndThreadIdPatternConverter] for keyword [aProcessId] ch.qos.logback.core.util.IncompatibleClassException
10:48:41.254 documentid-service> 	at ch.qos.logback.core.util.IncompatibleClassException
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:58)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.util.OptionHelper.instantiateByClassName(OptionHelper.java:45)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.util.OptionHelper.instantiateByClassName(OptionHelper.java:34)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.pattern.parser.Compiler.createConverter(Compiler.java:104)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.pattern.parser.Compiler.compile(Compiler.java:63)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.pattern.parser.Parser.compile(Parser.java:87)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.pattern.PatternLayoutBase.start(PatternLayoutBase.java:84)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.access.PatternLayout.start(PatternLayout.java:210)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.access.PatternLayoutEncoder.start(PatternLayoutEncoder.java:26)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.action.NestedComplexPropertyIA.end(NestedComplexPropertyIA.java:161)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.Interpreter.callEndAction(Interpreter.java:309)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.Interpreter.endElement(Interpreter.java:193)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.Interpreter.endElement(Interpreter.java:179)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.EventPlayer.play(EventPlayer.java:62)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:165)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:152)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:110)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:53)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.access.tomcat.LogbackValve.configureAsResource(LogbackValve.java:219)
10:48:41.254 documentid-service> 	at 	at ch.qos.logback.access.tomcat.LogbackValve.startInternal(LogbackValve.java:149)

We can't use the classic converter. But I have created new modules with converters for each log back event. One for LoggingEveent and another one for AccessEvent as per your comments.